### PR TITLE
changed conflict files

### PR DESCRIPTION
**Pull request by ViraMaximets**

### CODE REVIEW

File: src/main/java/com/softserve/app/controller/MainPageController.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+
+@Controller
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/MainPageController.java:15]: @RestController

File: src/main/java/com/softserve/app/controller/MainPageController.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.List;
+
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/MainPageController.java:14]: @AllArgsConstructor

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 3]
```diff
@@ -1,32 +1,78 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:10]: change * to .

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 4]
```diff
@@ -1,32 +1,78 @@
...
+            return ResponseEntity.ok(usr);
+        }
+        return ResponseEntity
+                .status(HttpStatus.FORBIDDEN)
+                .body(SportHubConstant.AUTHENTICATION_EXCEPTION.getMessage());
+    }
+
+
+    // update profile
+    @RequestMapping(method = RequestMethod.POST)
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:49]: @PostMaping

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 5]
```diff
@@ -1,32 +1,78 @@
...
 
-    // todo: add methods
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+
+    // show  profile
+    @RequestMapping(method = RequestMethod.GET)
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:32]: @PostMaping

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 6]
```diff
@@ -1,32 +1,78 @@
...
+    @ResponseBody
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:33]: delete

File: src/main/java/com/softserve/app/models/User.java [hunk 7]
```diff
@@ -1,59 +1,46 @@
...
-import javax.persistence.Enumerated;
-import javax.persistence.FetchType;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.JoinTable;
-import javax.persistence.ManyToMany;
-import javax.persistence.OneToMany;
+import javax.persistence.*;
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/models/User.java:6]: * to .

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 8]
```diff
@@ -1,32 +1,70 @@
...
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:10]: * заміни на всі через . (якщо що в настройках тре фун замін вимкути)

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 9]
```diff
@@ -1,32 +1,70 @@
...
+import org.springframework.web.multipart.MultipartFile;
 
 import java.util.Objects;
 
 
 @RestController
+@CrossOrigin(origins = "http://localhost:4200")
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:17]: localhost... заміни на *

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 10]
```diff
@@ -1,32 +1,70 @@
...
+        if (Objects.equals(userService.getCurrentUser(), usr)) {
+            return ResponseEntity.ok(userService.updateUser(userDto));
+        }
+        return ResponseEntity
+                .status(HttpStatus.FORBIDDEN)
+                .body(SportHubConstant.AUTHENTICATION_EXCEPTION.getMessage());
+    }
+
+
+    @RequestMapping(value = "/avatar", method = RequestMethod.POST)
```
Comment (nazar-goncharenko) [inline src/main/java/com/softserve/app/controller/UserController.java:57]: @PostMaping

File: src/main/java/com/softserve/app/controller/MainPageController.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+@RestController
+@AllArgsConstructor
+public class MainPageController {
+
+    private final UserService userService;
+
+
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    @ResponseBody
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/MainPageController.java:23]: if you use RestController annotation, you don't have to use ResponseBody annotation, please remove it. Change RequestMapping to GetMapping

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 12]
```diff
@@ -1,32 +1,78 @@
...
-
     // todo: user registration and login
 
-
     private final UserService userService;
 
-    // todo: add methods
 
+    @GetMapping
+    public ResponseEntity<Object> showProfile(
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/UserController.java:36]: return only DTO, if you want to return some status. Throw your custom exception in service with your http code

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 13]
```diff
@@ -1,32 +1,78 @@
...
+    @PostMapping
+    public ResponseEntity<Object> updatePersonal(
+            @RequestBody UserDTO userDto, @PathVariable Long user_id) {
+
+        User usr = userService.findById(user_id);
+
+        if (Objects.equals(userService.getCurrentUser(), usr)) {
+            return ResponseEntity.ok(userService.updateUser(userDto));
+        }
+        return ResponseEntity
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/UserController.java:59]: same as above

File: src/main/java/com/softserve/app/controller/UserController.java [hunk 14]
```diff
@@ -1,32 +1,78 @@
...
+
+    @RequestMapping(value = "/avatar", method = RequestMethod.POST)
+    public ResponseEntity<Object> updateAvatar(
+            @RequestParam("file") MultipartFile userAva,
+            @PathVariable Long user_id) {
+        User usr = userService.findById(user_id);
+
+        if (Objects.equals(userService.getCurrentUser(), usr)) {
+            return ResponseEntity.ok(userService.updateAvatar(user_id, userAva));
+        }
```
Comment (Qvel) [inline src/main/java/com/softserve/app/controller/UserController.java:73]: same as above

**Pull request closed without merging**